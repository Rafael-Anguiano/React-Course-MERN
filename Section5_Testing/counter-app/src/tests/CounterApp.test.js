import React from 'react';
import CounterApp from '../CounterApp';
import {shallow} from 'enzyme'
import '@testing-library/jest-dom'

describe('Tests of CounterApp component', () => {
    let wrapper = shallow(<CounterApp value={10} />);

    beforeEach(() => {
        wrapper = shallow(<CounterApp value={10} />)
    })

    test('should show CounterApp correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
    test('Value should match with the sent value', () => {
        const wrapper = shallow( <CounterApp value={100} />)
        
        const shownValue = wrapper.find('h2').text().trim()
        expect(shownValue).toBe(`100`);
    })
    
    test('should increment with counter +1', () => {
        wrapper.find('button').at(0).simulate('click');
        
        const counterText = wrapper.find('h2').text().trim();
        expect(counterText).toBe('11')
    })
    test('should decrement with counter -1', () => {
        wrapper.find('button').at(1).simulate('click');
        
        const counterText = wrapper.find('h2').text().trim();
        expect(counterText).toBe(`9`)
    })
    test('should reset to default value with button', () => {
        const wrapper = shallow( <CounterApp value={105} />)
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(2).simulate('click');
        const counterText = wrapper.find('h2').text().trim();
        expect(counterText).toBe(`105`)
    })
    
})
