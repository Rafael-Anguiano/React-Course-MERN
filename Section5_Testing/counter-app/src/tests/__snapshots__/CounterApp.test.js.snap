// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests of CounterApp component should show CounterApp correctly 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
</Fragment>
`;
